File: src/main/java/com/github/pnemonic/game/lottery/LotException.java
package com.github.pnemonic.game.lottery;

/**
 * Error with the lottery lot.
 *
 * @author moshew
 */
public class LotException extends IndexOutOfBoundsException {

    /**
     * Creates a new exception.
     */
    public LotException() {
        super();
    }

    /**
     * Creates a new exception.
     *
     * @param message the detail message.
     */
    public LotException(String message) {
        super(message);
    }

}

File: src/main/java/com/github/pnemonic/game/lottery/LotException.kt
package com.github.pnemonic.game.lottery

/**
 * Error with the lottery lot.
 *
 * @author moshew
 */
class LotException : IndexOutOfBoundsException {
    /**
     * Creates a new exception.
     */
    constructor() : super()

    /**
     * Creates a new exception.
     *
     * @param message the detail message.
     */
    constructor(message: String) : super(message)
}